src/runtime_src/core/pcie/linux/system_linux.cpp

```diff
@@ -36,22 +38,32 @@
 
 namespace {
 
+xrt_core::system_linux* singleton = nullptr;
+
 // Singleton registers with base class xrt_core::system
-// during static global initialization.  If statically
+// during static global initialization. If statically
 // linking with libxrt_core, then explicit initialiation
 // is required
 static xrt_core::system_linux*
-singleton_instance()
+singleton_system_linux()
```

Comment:
Let's rewrite this function a bit tomorrow.  It is obfuscated from all the comments right now, but doesn't look quite right.

Reply:
Ha! I thought I followed the sample code in system class...

Reply:
You don't need to change anything in regards to construction of the system singleton.

Reply:
As discussed offline, we'll do it this way for now until we have our driver plug-in implemented.